docs/DeveloperGuide.md

```diff
@@ -1,13 +1,249 @@
 # Developer Guide
 
+## Table of contents
```

Comment:
Good use of indentation to organize the Table of Contents.
However, since there are 3 types of indentation, could there be a better way of labelling each line? say 1.1.1?
Just a small suggestion.